pkg/server/container_create.go

```diff
@@ -153,6 +154,18 @@ func (c *criService) CreateContainer(ctx context.Context, r *runtime.CreateConta
 		return nil, errors.Wrapf(err, "failed to generate container %q spec", id)
 	}
 
+	meta.ProcessLabel = spec.Process.SelinuxLabel
+	if config.GetLinux().GetSecurityContext().GetPrivileged() {
+		// If privileged don't set the SELinux label but still record it on the container so
+		// the unused MCS label can be release later
+		spec.Process.SelinuxLabel = ""
+	}
```

Comment:
From what I understand this mimics docker and cri-o behavior, but as far as I know the purpose is to create a "super" process label which effectively covers all MCS categories. I think this may be helpful for file access in some situations.

Reply:
Setting spec.Process.SeLinuxLabel = "" basically means, don't modify the selinux label, it will just inherit.  This is the historical behavior of --privileged.  If privileged=true then don't mess with selinux labels of the process.

Reply:
Inherit from what? The pod or the system default?

For MCS, I could see equating "privileged" with "the most privileged level" and setting `s0-s0:c0.c1023` so the process ended up with a dominating label, relative to other containers in the pod.

The most generous interpretation would be "ensure the process runs with the most privileged SELinux label available on the system" - something like Fedora's unconfined `spc_t`. That seems like it goes too far, though, and has other side effects - the privileged container may write files that other containers in the pod can't read.

It's easy enough to specify SELinux attributes at the pod/container level to get either of those behaviors - it's a one-line change assuming the "inheritance" model you mentioned - so hard-coding a particular behavior may be more trouble than it's worth, unless it's already a de-facto standard.

Reply:
At this point in the code I think we can say we are inheriting from the system default, that of the shim, as defined in container-selinux here: https://github.com/containers/container-selinux/blob/6caf15d8540f3f4bb4f6f1251e28bcfa3123896d/container.fc#L6
(or if we are outside of that location according to the file label on the executable, which will influence the running process label)

For these purposes we can say `s0`, `s0-s0`, and `s0-s0:c0.c1023` are equivalent.